### PR TITLE
Use readonly structs where possible.

### PR DESCRIPTION
**Pull request by grokys**
C# 7.2 introduced [readonly structs](https://docs.microsoft.com/en-us/dotnet/csharp/reference-semantics-with-value-types). Using this construct we can make sure that future modifications to these structs don't make them immutable and C# can generate more efficient code.

### CODE REVIEW

**Comment by grokys** [comment]
We're going to have to wait until Appveyor update their VS image for this.

**Comment by jp2masa** [comment]
In case you don't already know it, there's a `Visual Studio 2017 Preview` image in AppVeyor, which currently has VS 15.5 Preview 4 installed I think.

File: src/Avalonia.Visuals/Avalonia.Visuals.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Avalonia</RootNamespace>
+    <LangVersion>latest</LangVersion>
```
Comment (aelij) [inline src/Avalonia.Visuals/Avalonia.Visuals.csproj]: Isn't there a common build props file you can add it to? I think the same language version should be used throughout the entire solution.

Comment (grokys) [reply src/Avalonia.Visuals/Avalonia.Visuals.csproj]: Ah yeah good point!

**Comment by grokys** [comment]
@aelij @jkoritzinsky updated the travis .net SDK to one that supports this feature, and moved the `<LangVersion>` to a shared .props file. CI is now passing.

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**